components/ca-form.js

```diff
@@ -1094,6 +1094,8 @@ define([
             if (format === 'radio') {
 
                 el = createElementLegacy(null, 'ca-radio-group', {});
+                el.data = schema.extends[0];
```

Comment:
Only handles one particular form of enum declaration in JSON schema. Which is fine but it isn't a drop-in for ca-select. Note this code in ca-form.createList() - which works with ca-select - and is the fuller version of what you have:

```
                // if the schema is extended
                if (schema.extends && schema.extends[0]) {
                    // and the extended schema contains a reference
                    if (schema.extends[0].$ref) {
                        // then use this reference to set the "src" property in order to resolve the reference
                        attrs.src = schema.extends[0].$ref;
                    } else {
                        // otherwise use the extended schema
                        attrs.schema = schema.extends[0];
                    }
                } else {
                    // schema is not extended so simply use the base schema
                    attrs.schema = schema;
                }
```
and this code from ca-select.schema:
```
set schema(schema) {

            const selectData = [];
            const data = schema.enum || schema.type;

            if (Array.isArray(data) && data.length) {

                for (let i = 0, l = data.length; i < l; i++) {

                    const value = (data[i].enum && data[i].enum[0]) || data[i];
                    const label = data[i].title || (l === 1 && schema.title) || data[i];
                    let links = data[i].links || (l === 1 && schema.links) || [];
                    links = links.filter(link => link.rel === 'preview');

                    selectData.push({
                        value,
                        label,
                        attachment: (links.length && links[0]) || null
                    });
                }
            }

            this.data = selectData;
        }
```

To do it properly, we'd be refactoring these complex conditions into some kind of common code and then using for both controls.